src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -516,12 +516,14 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
 
         case NI_VectorT_ConvertToDouble:
         {
+#ifdef TARGET_AMD64
```

Comment:
Shouldn't this be `#ifdef TARGET_X86` and filter out the conversion from `int64_t` (since that requires `Avx512F.X64`?

Same goes for the others below.

Reply:
Hi Tanner,

Thanks for the comment. I am trying to understand the behavior on x86 machines. Can you explain your comment further as to why do we not want ```TARGET_X86``` to use AVX512?. I understood that we cannot do ```double -> long``` and '''double -> ulong``` since it needs AVX512F.X64 but why would that be the case for others i.e. ```float -> int``` and ```float -> uint```. 

![Capture2](https://github.com/dotnet/runtime/assets/13829923/1e2e9b9e-6d71-417d-b140-f7e1e9685223)
![Capture1](https://github.com/dotnet/runtime/assets/13829923/01c05e6b-2092-463a-8ad2-55aefcd6b79e)
![Capture](https://github.com/dotnet/runtime/assets/13829923/fee5ee11-a39b-4369-8cbf-fb58fc4110de)

I see from the Intrinsic Manual that ```x86``` will behave as if W0 version of conversion instruction is used.

As of now, we are trying to understand the behavior of ```x86``` and how is the code structured for helper functions.

Reply:
We do want this to work for x86 and x64.

For 32-bit inputs, this works great as the source or destination is a 32-bit general purpose register.

For 64-bit inputs, the relevant instruction encoding is only available for "long mode" (64-bit processes) since it needs a 64-bit general purpose register.

With how you have the code now, we are only accelerating for 64-bit processes. The change I suggested allows 32-bit processes to also work, provided the instruction won't require a 64-bit register

Reply:
Hi Tanner,

Is there a reason why we have this comment in ```jithelpers.h```. 
```
    // CORINFO_HELP_DBL2INT, CORINFO_HELP_DBL2UINT, and CORINFO_HELP_DBL2LONG get
    // patched for CPUs that support SSE2 (P4 and above).
```
I see that the above 3 helper functions are defined as DYNAMICJITHELPER and have a different implementation from that in ```jithelpers.cpp```. Was there a reason why we had different implementation in ```jithelp.asm``` specifically for ```X86```? This is becvause I am thinking of moving everything to ```jithelpers.cpp``` but I dont know why it was done in the first place.

Reply:
That I'm not sure, @BruceForstall or @jkotas might be more familiar with the historical reason for that split here.

Reply:
> Was there a reason why we had different implementation in jithelp.asm specifically for X86? 

It is an optimization from the days when 32-bit x86 was the platform that we have been optimizing the most, and when SSE2 was not in the baseline. I do not see a problem with deleting all that (it may be a good idea to do it in a separate PR).

Reply:
Thanks @jkotas . This PR actually modifies all the code related to that optimization and will touch almost all the same places in code. I will see if removing it makes sense in this PR since things are already in place for ```AMD64```. This PR is about optimizing the helpers in ```jithelpers.cpp``` for Intel ISA. Since we are already touching helpers for ```AMD64```, it makes sense to change this code here. This PR will otherwise introduce and require extra effort for ```X86``` which will then be removed shortly due to the changes we are suggesting. Let me know if this sounds okay.

Reply:
> I do not see a problem with deleting all that (it may be a good idea to do it in a separate PR).

FYI #98858 does that already.

Reply:
@MichalPetryka How is the timeline looking for https://github.com/dotnet/runtime/pull/98858 ? Any idea how much time will it take to get all tests passing and merge it? If we are to pickup changes from that PR, it will be better if they are merged since this PR changes a lot of code around the similar places as PR [98858](https://github.com/dotnet/runtime/pull/98858)

Reply:
> How is the timeline looking for #98858 ?

It's pretty much done I think, the only possible concern are the 4 unused R2R helpers which I've ported too cause they're simpler in managed but I'm not 100% sure if what I did will keep them R2R compatible (wouldn't be a concern if we decide to bump the R2R version here).

> get all tests passing

Failures there seem unrelated, I'm gonna merge main to have the CI rerun pipelines there.

Reply:
Thanks @MichalPetryka for the update. We have started building our changes on top of your PR to make sure we are incorporating those changes. Thanks for catching our PR early in the process.

Reply:
> Avx512F.X64

Hi Tanner,

For the below image, can you explain why will it not support 32 bit mode for conversion to double from ulong. This is with respect to your comment https://github.com/dotnet/runtime/pull/97529#discussion_r1506978162

![Capture3](https://github.com/dotnet/runtime/assets/13829923/7a10c635-659b-4f54-921e-dac399b9b88a)

Reply:
It's the scalar form that won't work. The vector form works on 32-bit just fine.

Reply:
@MichalPetryka Is the other PR close to being merged? I have a separate branch(https://github.com/dotnet/runtime/pull/99143/checks) where I merged both this and your changes and made some other required changes on top. I'm waiting for your PR to be merged before we can update this PR

Reply:
Thanks @tannergooding for the help.. I was able to verify the changes on x86 yesterday.

Reply:
@tannergooding the changes you suggested were made to enable AVX512F on x86 as well.